watchtower-plugin/src/retrier.rs

```diff
@@ -647,13 +647,17 @@ mod tests {
         add_appointment_receipt.sign(&tower_sk);
         let add_appointment_response =
             get_dummy_add_appointment_response(appointment.locator, &add_appointment_receipt);
-        let api_mock = server.mock(|when, then| {
-            when.method(POST).path(Endpoint::AddAppointment.path());
-            then.status(200)
-                .delay(Duration::from_secs_f64(API_DELAY))
-                .header("content-type", "application/json")
-                .json_body(json!(add_appointment_response));
-        });
+
+        let api_mock = server
+            .mock("POST", Endpoint::AddAppointment.path().as_str())
+            .with_status(200)
+            .with_header("content-type", "application/json")
+            .with_body_from_fn(move |w| {
+                thread::sleep(Duration::from_secs_f64(API_DELAY));
```

Comment:
I think it has to do with the mock having a synchronous wait. However, the function does not allow async closures atm.

I PRed a change to add delays to the repo but the maintainers seemed not to like it on the way it was (standalone, instead of trough something like this method), so I may rework it so async closures are accepted (I'm still debating about it).